### PR TITLE
view: permission check, section display

### LINKED ISSUE #240: [BUG] Domain admin: Identify and access > Roles should not be visible
**Describe the bug**
'Identify and access > Roles' screen is visible but the domain admin has no access there so it doesn't make sense.

**To Reproduce**
1. Log in as domain admin, go to 'Identify and access > Roles'

**Expected behavior**
'Identify and access > Roles' should not be visible when logged as domain admin.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Browser: Chrome
 - Version: 80.0.3987.132 (Official Build) (64-bit)

**Additional context**
None.

### LINKED ISSUE #241: [BUG] Domain admin: Offerings - more options than in the old UI
**Describe the bug**
No Offerings menu options are hidden, as they are in the old UI.

**To Reproduce**
1. Log in as domain admin, go to 'Offerings' - all options are visible there.

**Expected behavior**
Only two options should be visible, as in the old UI - compute and disk offerings.

**Screenshots**
New UI:
image

Old UI:
image

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Browser: Chrome
 - Version: 80.0.3987.132 (Official Build) (64-bit)

**Additional context**
Offerings > System Offerings generates a 404 error, as mentioned in #233.

### LINKED ISSUE #244: [BUG] Normal user: the whole Offerings section should not be visible
**Describe the bug**
The 'Offerings' section is visible, but mostly unusable, due to missing permissions. 

**To Reproduce**
1. Go to 'Offerings', try some of the sub-sections.

**Expected behavior**
The whole 'Offerings' section must be hidden, like in the old UI.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Browser: Chrome
 - Version: 80.0.3987.132 (Official Build) (64-bit)

**Additional context**
None.

### PR DESCRIPTION
**Pull request by shwstppr**
Changes permission checking behaviour.
Multiple APIs can be passed in section child's permission property as an array.
First API will be used to compute the route for the child sub-section.
Remaining APIs will only be used to verify permissions. If any of the passed API is not found in the allowed APIs for logged-in user's role section/sub-section won't be displayed.

Fixes #241 
Fixes #244 
Fixes #240

### CODE REVIEW

**Comment by shwstppr** [comment]
@rhtyd please review. Do you think to check multiple APIs for permission a better option for checking permission and visibility for sections when the primary API doesn't have allowed roles defined

File: src/store/modules/permission.js [hunk 1]
```diff
@@ -20,11 +20,12 @@ import { asyncRouterMap, constantRouterMap } from '@/config/router'
 function hasApi (apis, route) {
   if (route.meta && route.meta.permission) {
     for (const permission of route.meta.permission) {
-      if (apis.includes(permission)) {
-        return true
+      console.log(route.meta.name, permission)
```
Comment (rhtyd) [inline src/store/modules/permission.js:23]: @shwstppr could this make permissions too strict; we've use this to show views when we want any of the APIs allowed in the list. Let's discuss tomorrow.

**Comment by rhtyd** [comment]
@shwstppr I've followed your change, which required config wide refactoring.

**Comment by shwstppr** [comment]
@rhtyd even this would fail for read-only admin as it won't show Offerings section for that,
Screenshot from 2020-05-07 10-42-01

**Comment by rhtyd** [comment]
@shwstppr fixed on latest master now

**Pull request merged**